presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/SchemaAccessControlRule.java

```diff
@@ -24,26 +24,42 @@
 public class SchemaAccessControlRule
 {
     private final boolean owner;
+    private final Optional<AccessMode> allow;
     private final Optional<Pattern> userRegex;
     private final Optional<Pattern> schemaRegex;
 
     @JsonCreator
     public SchemaAccessControlRule(
             @JsonProperty("owner") boolean owner,
+            @JsonProperty("allow") Optional<AccessMode> allow,
             @JsonProperty("user") Optional<Pattern> userRegex,
             @JsonProperty("schema") Optional<Pattern> schemaRegex)
     {
         this.owner = owner;
+        this.allow = requireNonNull(allow, "allow is null");
         this.userRegex = requireNonNull(userRegex, "userRegex is null");
         this.schemaRegex = requireNonNull(schemaRegex, "sourceRegex is null");
     }
 
-    public Optional<Boolean> match(String user, String schema)
+    public Optional<Boolean> isSchemaOwner(String user, String schema)
     {
         if (userRegex.map(regex -> regex.matcher(user).matches()).orElse(true) &&
                 schemaRegex.map(regex -> regex.matcher(schema).matches()).orElse(true)) {
             return Optional.of(owner);
         }
         return Optional.empty();
     }
+
+    public Optional<AccessMode> match(String user, String schema)
+    {
+        if (userRegex.map(regex -> regex.matcher(user).matches()).orElse(true) &&
+                schemaRegex.map(regex -> regex.matcher(schema).matches()).orElse(true)) {
+            // by default, if user did not defined visibility, assume schema is visible for back compatible
```

Comment:
Remove the comment after applying the suggestion in the constructor.

Reply:
ok